overlord/snapstate/handlers.go

```diff
@@ -977,15 +977,20 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func snapdSnapInstalled(st *state.State) bool {
+	var snapst SnapState
+	err := Get(st, "snapd", &snapst)
+	return err == nil
+}
+
 // maybeRestart will schedule a reboot or restart as needed for the
 // just linked snap with info if it's a core or snapd or kernel snap.
 func maybeRestart(t *state.Task, info *snap.Info) {
 	st := t.State()
 
-	// TODO: once classic uses the snapd snap we need to restart
-	//       here too
 	if release.OnClassic {
-		if info.Type == snap.TypeOS {
+		if (info.Type == snap.TypeOS && !snapdSnapInstalled(st)) ||
```

Comment:
While this doesn't seem to be a new property, looking at this code I was thinking that we should check if a snap is a re-execution target. After all, if re-execution is disabled then restart is pointless.

Reply:
Yeah, good point. Given that 2.38 is close I would like to do that as a followup though.